tests/queryEditor.spec.ts

```diff
@@ -135,3 +135,35 @@ test('"Timeseries custom query" multiple ts OK', async ({ selectors, readProvisi
   const tsWithUnits = tsExternalIds.map((ts, i) => `${ts}-${units[i]}`);
   await expect(panelEditPage.panel.fieldNames).toContainText(tsWithUnits);
 });
+
+
+test('"Event query" as table is OK', async ({ page, gotoDashboardPage, readProvisionedDashboard, grafanaVersion }) => {
+  const dashboard = await readProvisionedDashboard({ fileName: 'weather-station.json' });
+  const dashboardPage = await gotoDashboardPage(dashboard);
+  
+  const panelEditPage = await dashboardPage.gotoPanelEditPage('3')
+
+  await expect(panelEditPage.panel.fieldNames).toContainText(["externalId", "description", "startTime", "endTime"]);
+
+  const query = `
+    {
+      "prefix": {
+          "property": ["externalId"],
+          "value": "test_event (1"
+      }
+    }`;
+
+  if (semver.gte(grafanaVersion, '10.2.0')) {
+    await page.getByTestId(/Code editor container/).getByRole("textbox").first().fill(query, { force: true });
+  } else {
+    await page.getByLabel(/Code editor container/).getByRole("textbox").first().fill(query, { force: true });
+  }
+
+  await waitForQueriesToFinish(page, grafanaVersion);
+  await expect(panelEditPage.refreshPanel({ waitForResponsePredicateCallback: isCdfResponse('/events/list') })).toBeOK();
+
+  await expect.poll(async () => {
+    const cellTexts = await panelEditPage.panel.data.getByRole('cell').allInnerTexts();
+    return cellTexts.every(text => /test_event \(0/.test(text)) && !cellTexts.some(text => /test_event \(1-9/.test(text));
+  }, { timeout: 10000 }).toBeTruthy();
```

Comment:
can we add a comment describing what this is checking?

Reply:
added